shepherd/shepherd/shepherd.py

```diff
@@ -104,6 +108,10 @@ def slaughter_sheep(self, sheep_id: str) -> None:
         :return:
         """
         logging.info('Slaughtering sheep `%s`', sheep_id)
+
+        for greenlet in self._sheep_greenlets[sheep_id]:
```

Comment:
Not sure if you want to kill the greenlets here. If it is the case, then you gonna need to start them every time you kill and (re)start the respective sheep.

Reply:
You're right, thanks.